### PR TITLE
feat(crons-db): Stop accessing `MonitorEnvironment.environment` directly

### PR DESCRIPTION
**Pull request by wedamija**
We want to move all crons related tables to a separate database. To facilitate this, we need to stop accessing the `MonitorEnvironment.environment` relation, and use a helper function instead.

### CODE REVIEW

File: src/sentry/monitors/models.py [hunk 1]
```diff
@@ -623,8 +623,15 @@ class Meta:
 
     __repr__ = sane_repr("monitor_id", "environment_id")
 
+    def get_environment(self) -> Environment:
+        return Environment.objects.get(self.environment_id)
```
Comment (wedamija) [inline src/sentry/monitors/models.py:627]: Probably should fetch this from the cache

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/sentry/monitors/models.py [hunk 2]
```diff
@@ -623,8 +623,15 @@ class Meta:
 
     __repr__ = sane_repr("monitor_id", "environment_id")
 
+    def get_environment(self) -> Environment:
+        return Environment.objects.get(id=self.environment_id)
```
Comment (markstory) [inline src/sentry/monitors/models.py:627]: You might want to use cached reads here as environments don't change that often. The `Environment` model will need to be changed to support caching first though.

**Review by markstory — APPROVED** [review]

**Review by cmanallen — APPROVED** [review]
Other than Mark's comment on caching this looks good.

**Comment by wedamija** [comment]
Oops, thought I pushed my changes with the caching here, I'll put up a separate pr

**Comment by wedamija** [comment]
https://github.com/getsentry/sentry/pull/65475

**Pull request merged**